src/scala/com/github/johnynek/bazel_deps/DepsModel.scala

```diff
@@ -182,18 +182,33 @@ object ArtifactOrProject {
   }
 }
 
-case class Sha1Value(toHex: String)
+sealed trait DigestType {
+  def getDigestInstance: MessageDigest
+  def expectedLength: Int
+}
+object DigestType {
+  case object Sha1 extends DigestType {
+    def getDigestInstance: MessageDigest = MessageDigest.getInstance("SHA-1")
+   def expectedLength: Int = 40
+  }
+  case object Sha256 extends DigestType {
+    def getDigestInstance: MessageDigest = MessageDigest.getInstance("SHA-256")
+    def expectedLength: Int = 64
+  }
+}
 
-object Sha1Value {
+case class ShaValue(toHex: String, digestType: DigestType)
 
-  def computeShaOf(f: File): Try[Sha1Value] = Try {
-    val sha = MessageDigest.getInstance("SHA-1")
+object ShaValue {
+
+  def computeShaOf(digestType: DigestType, f: File): Try[ShaValue] = Try {
+    val shaInstance = digestType.getDigestInstance
```

Comment:
can we move this into the `try` just in to be extra paranoid?

Reply:
happy to, but its also covered in the `Try{}` block, so should be ok i think